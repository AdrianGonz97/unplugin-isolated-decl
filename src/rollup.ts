import { IsolatedDecl } from './index'

export default IsolatedDecl.rollup as typeof IsolatedDecl.rollup
