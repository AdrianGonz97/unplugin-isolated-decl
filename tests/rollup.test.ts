import path from 'node:path'
import { expect, test } from 'vitest'
import { rollup } from 'rollup'
import UnpluginIsolatedDecl from '../src/rollup'

test('rollup', async () => {
  const input = path.resolve(__dirname, 'fixtures/main.ts')
  const dist = path.resolve(__dirname, 'temp')

  const bundle = await rollup({
    input,
    plugins: [UnpluginIsolatedDecl(), { name: 'raw', transform: () => '' }],
    logLevel: 'silent',
  })

  const result = await bundle.generate({
    dir: dist,
  })
  expect(
    result.output.map(
      (asset) =>
        `// ${asset.fileName}\n${asset.type === 'chunk' ? asset.code : asset.source}`,
    ),
  ).toMatchSnapshot()
})
