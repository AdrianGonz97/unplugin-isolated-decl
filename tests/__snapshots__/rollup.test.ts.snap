// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup 1`] = `
[
  "// main.js
function hello(s) {
  return "hello" + s;
}
let num = 1;

export { hello, num };
",
  "// main.d.ts
import { type Num } from "./types";
export type Str = string;
export declare function hello(s: Str): Str;
export declare let num: Num;
",
  "// types.d.ts
export type Num = number;
",
  "// types2.d.ts
export type Num2 = number;
",
]
`;
